### PR TITLE
fix body offsetTop value issue

### PR DESCRIPTION
**Pull request by gbujanski**
Regarding #31569 &  #31950

Just like Mikołaj wrote: "scrollSpy reads the wrong window offsetTop and marks the wrong link as active".  This should fix it.

### CODE REVIEW

**Comment by XhmikosR** [comment]
Thanks for the PR, but we need a test case which would fail on main and pass here to make sure the patch is right :)

**Comment by xuyihan-china** [comment]
bootstrap's scrollSpy's API document is difficult to use ,please improve

File: js/src/dom/manipulator.js [hunk 1]
```diff
@@ -62,8 +62,8 @@ const Manipulator = {
     const rect = element.getBoundingClientRect()
 
     return {
-      top: rect.top + document.body.scrollTop,
-      left: rect.left + document.body.scrollLeft
+      top: rect.top + (document.body.scrollTop || document.documentElement.scrollTop),
```
Comment (emilio) [inline js/src/dom/manipulator.js:65]: I think you really want `document.scrollingElement.scrollTop`.

**Comment by XhmikosR** [comment]
@gbujanski can you add the needed test and also check out @emilio's comment?

**Comment by XhmikosR** [comment]
Closing due to lack of response. You can comment here and we can re-open the PR, if you decide to resume working on the requested changes.

*This is an automated reply*

**Pull request closed without merging**